### ISSUE TITLE
Backtest stats are different on short and long period

### ISSUE THREAD

**Issue by MohamedDhouib** [freqtrade/freqtrade#8496]
## Describe your environment

  * Operating system: windows
  * Python Version: 3.8 (`python -V`)
  * CCXT version: ccxt==3.0.50 (`pip freeze | grep ccxt`)
  * Freqtrade Version: freqtrade 2022.10 (`freqtrade -V` or `docker compose run --rm freqtrade -V` for Freqtrade running in docker)
  
Note: All issues other than enhancement requests will be closed without further comment if the above template is deleted or not filled out.

## Describe the problem:

 i'm having a difference between the result of 2 backtest stats generated on freqtrade and i don't have an explanation.

I have a strategy having a performance of +200% in December 2023 (screenshot attached) but when i ran a full backtest since 2022 with monthly breakdown i have the exact same stats (winnig trade/lossing trades etc) but the monthly perf in % is far from +200%. It's actually +16.59% according to the monthly breakdown (see Excel screnshot). What is there such a huge difference? 

Max open trades: 1
Static list: 80 pairs

December 2022 backtest only:
![image](https://user-images.githubusercontent.com/56609682/231835563-fbfb11a7-9c36-4c37-9dca-b0f86af4d497.png)

December stats on 2022 yearly backtest:
![image](https://user-images.githubusercontent.com/56609682/231835748-827cea45-d5fa-4a69-9148-8cfc890671ca.png)
![image](https://user-images.githubusercontent.com/56609682/231835723-31eda8bc-14f0-461e-836f-22a971ebf34b.png)

**Comment by MohamedDhouib** [comment]
I have to add that "method": "ShuffleFilter" is deactivated so even when i run the backtest multiple times i have the same result to the 4th decimal.

**Comment by xmatthias** [comment]
All you're **really** seeing is that it's roughly the same position amount (nothing more, nothing less).

it could be different pairs it's trading, or also running into the "upper" limit of the position (i doubt exchanges allow order-sizes > 100M - at least not on smaller coins).

You limit the bot to 1 parallel trade (it seems) - which can result in a significant portion of your wallet being on the sidelines (you enter with 50M - but the account would be 300M - which leaves 250M on the "sidelines"). 
So the growth on the account is smaller.

Another huge effect can be the occupation of trade slots.
if the very first trade is different due to another trade still being open since November, it can change every single trade in December (even though the number is the same, it's all 1 candle later, on different pairs (for example) - but this other pair behaved differently.

overall - i'd view your results VERY skeptical. You turn 1000$ into 500M over the course of a year - which is a pretty clear sign that you're exploiting one of the [backtesting assumptions](https://www.freqtrade.io/en/stable/backtesting/#assumptions-made-by-backtesting) - or that it's looking into the future (backtesting allows that - but live won't).

**Comment by MohamedDhouib** [comment]
Thank's for your reponse Mattias.

From your response the only possible explanation is indeed the significant portion traded on exchnage but does freqtrade take that into consideration? Meaning is there a threshold of the max amount traded based on the exchange?
"max_open_trades": 1,
"stake_amount": "unlimited",
"tradable_balance_ratio": 1.0
 "method": "ShuffleFilter" is deactivated

I did the same thing for January and i'm having the same issue:
Only explanation here is that one of them has exlusive rule (last day of month is not included or something like that).

only january:
![image](https://user-images.githubusercontent.com/56609682/231847070-12927898-cb20-484a-b34e-d296e7e1647a.png)

Month break down:
![image](https://user-images.githubusercontent.com/56609682/231847228-4ab17f14-5c0f-4a94-9a95-02e64bce2b4b.png)

For your last comment about the backtest, i have a very low trailing offset (0.1%) which is really hard to get using the freqtrade bot because i think it's not designed for very high frequency on multiple pairs. I coded it from scratch and i have a 200ms scan frequency of assets prices which allows me to be able to exit on very tight trailing offsets (I catch every single wick exit for now). for now i match the backtest after few weeks of live exec (freqtrade dry run is very different due to it's slowness scanning prices).

**Comment by xmatthias** [comment]
Backtesting with low trailing offset will not work - and is exploiting the way backtesting works.

If you think about the assumptions very carefully (and their sequence, and what that means for a tight stoploss) - you'll realize that the trailing stop will hit 0.1% below the candle's high.
that's not realistic given regular candle patterns - therefore your whole backtest will not work, and is flawed, showing extensive profits where they are not.

your best bet will be to look at each trade in detail to determine what's causing this.
it's most likely shifted trades - which cause strongly different results.

**Comment by MohamedDhouib** [comment]
I agree that using tight trailing does exploit the backtest but only in some specific cases that w can clearly identify:

1. If SL and Positive Trailing SL on the same candle, in this case I agree that it's a clear backtest exploit coz we don't know which one was hit first
2. If open and Trailing SL on the same candle, we woudln't know which one was hit first. price can drop to entry passing by the trailing and it may coz some confusion

**I'm not trying to force my pov**, only trying to understand why is it impossible to chase the 0.1% drop if i'm not in the previous bullets (1. & 2.). If im trading on the 5min TF and my trailing exit avg duration is 1h (faar from my entry candle) and i'm using the 1m time-detail in the backtest, how can the backtest be false? can you please give one example to illustrate any case in which it can cause an issue?
On the discord channel, members are saying that it's impossibe to get that wick and exit on the trailing so it's purely technical and i've solved that issue by coding it from scratch as i said in my previous comment. Now when you say that the backtest itself is false, that's a big concern....

**Comment by xmatthias** [comment]
you might not be trying to intentionally - but you still are by using such a tight trailing stop.
if you search through issues - you'll probably find 10s of issues where people thought the same (at least it feels like i explained this 10s of times), just to be disappointed realizing that their logic is only exploiting backtesting.

---

independent on when the trailing stop is happening (on open candle - which btw has special handling), the following behavior will always happen.

Once trailing_stop_offset is reached - the stoploss will be adjusted (to the candle's high). 
With a trailing stop of 0.1% - it's almost guaranteed that the exit will be in the adjustment candle, and the trailing stop will happen at `high * (1-0.001)` (trailing_stop - or 0.1% - below the high).
The only case this would not happen is if the low-high difference is <0.1% - and that's effectively never the case - not even on big markets like ETH or BTC (smaller markets usually have a lot wider movement ranges).

While a move of "open - all the way to high - and then back" is surely possible - usual movements will be "open, a bit up, a bit down, maybe all they way down, then up" (also on 1m timeframes) - therefore the "reached high, therefore sell" is not as reliable as you think - as the trailing stop will trigger "any time above offset that's followed by a move of 0.1% down".

The trailing stoploss logic of using 0.1% therefore becomes a "high > offset - sell at high-0.1%". it's no longer a trailing stop, really - as the moment it starts to trail - it'll exit (with an inflated price).

your average duration of 1h will only mean one thing - your offset was reached after 1 hour.

---

most likely, you can do basically the same logic (roughly the same trade duration)  - but more reliable - by using your trailing_offset value (-0.1%?) as ROI.

The difference here is that ROI exits will not be "high - trailing-stop" value - but they'll be "open_rate * roi" - selling mid-candle (where your offset is actually reached) - instead of at high - trailing-stop.

**Comment by MohamedDhouib** [comment]
Thank you very muh for the detailled explanation. I understand your pov now.

Anyway i've done what you suggested with only ROI exits i still have the **exact same win rate ratio**:
![image](https://user-images.githubusercontent.com/56609682/232021384-9dd71ab2-a03e-41ee-8405-71def2e1a68f.png)

i have x20 difference with the trailing exits backtest on 1.5 year which seems logic for me because the ROI exists are 2% exacly while the trailing exits my go further that 2%.
I think that I do not have the problem you explained because my only exit signal on loss is the stop loss or the ROI. my stop loss is -20% so it's nearly impossible to have a candle that wicks both my trailing offset and my SL. and my ROI is time based only.

FYI i'm not talking out of theory only, i've been running: freqtrade dry run vs my implementation live exec vs backtest and dry run is quit far from backtest (frequency of assets scan doesn't let it get the trailing wicks) while my implmentation has only 1% difference with backtest on 3 weeks exec.

To sumup I still do not know why is there such a difference between monthly breakdown on long period and short period backtest, I think that short period backtest is more reliable, i'll get into the backtest file and breakdown everything per trade.
Thank you very much for your comments, helpful as usual 💯!

If you see anything that can make my backtest biased (in the backtest logic and not indicators/lookahead bias) even with only ROI/SL, please do not hesitate to tell me, worked so hard on the from scratch implementation, don't want it to be a backtest issue.

**Comment by xmatthias** [comment]
Well i'm not sure what your "from scratch implementation" entails.
As you've mentioned it a few times now, without actually sharing any details, i'll assume you're not going to share / contribute that.

That makes it irrelevant to me - and considering that many parts are shared, you might've messed up something in the backtesting logic as part of your modifications (i'm sure you'll tell me you didn't).

Now one thing for sure: 
i'm always only talking about the most recent version. your version is half a year old - while that shouldn't make a significant difference (from broad memory) - it's also not something i would rely on strongly.

Once you know ACTUAL details of the differences (my suspicion is too big stakes, leaving tons of stake unused) - it might be something actionable, IF it appears on a unmodified, uptodate freqtrade installation (either current stable, or develop).

**Comment by MohamedDhouib** [comment]
When I say from scratch I literraly mean from scratch. I dont even use ccxt nor binance python library to not loose milliseconds putting orders or getting data. I get data and put orders using binance api using http. for indicators i optimized it using parallelism so i take 1 second to scan all pairs+calculte indicators and get into the trade.

I updated my freqtrade version to the 2023.3. Same result.

I have no problem sharing my strategy structure because it's really simple and all of the complexity is in the way i use indicators.
Here's my code without my entry/exit conditions, i think it's enough to have an idea whteher i messed up with a parameter or not.
`

    class Freqtrade_HF(IStrategy):

      position_adjustment_enable = True
    
      trailing_stop = False
      trailing_stop_positive = 0.001
      trailing_stop_positive_offset = 0.02
      trailing_only_offset_is_reached = True
  
      use_sell_signal = True
      sell_profit_only = True
	  
      minimal_roi = {
          "0": 0.02,
          "200": -1
      }
  
      stoploss = -0.2
  
      timeframe = '5m'
      
      process_only_new_candles = True
  
  
      def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
  
          added very simple indictaors:
		  ema/sma/rsi with sevral lengths (10 in total)
  
          return dataframe
  
      def populate_buy_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
          conditions = []
  
          conditions.append(
              (   
                  condition 1 based only on dataframe data (indicators+ candles data)
                  
  
              )
          )
  
          conditions.append(
              (   
                  
  
                  condition 2 based only on dataframe data (indicators+ candles data)
                  
              )
			  
		  conditions.append(
              (   
                  
  
                  condition 3 based only on dataframe data (indicators+ candles data)
                  
              )
          )
  
          if conditions:
              dataframe.loc[
                  reduce(lambda x, y: x | y, conditions),
                  'buy'
              ]=1
          
          return dataframe
  
      def populate_sell_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
          conditions = []
  
          conditions.append(
              (   
			  
                  condition 1 based only on dataframe data (indicators+ candles data)
               
              )
              |
              (
                  condition 2 based only on dataframe data (indicators+ candles data)
               
              )
			  |
              (
                  condition 3 based only on dataframe data (indicators+ candles data)
               
              )
			  
              
          )
  
          if conditions:
              dataframe.loc[
                  reduce(lambda x, y: x | y, conditions),
                  'sell'
              ]=1
  
          
          return dataframe` 

Concerning the long term backtest breakdown vs short term backtest, i'll get into the data and let you know what's going on...

**Comment by froggleston** [comment]
As with most things like this, the devil is in the details. It's important to know exactly how you're using indicators and signals.

Also, with tight trailing, 1m timeframe detail may still be not enough resolution to accurately quantify what's happening. Tight trailing makes heavy assumptions on price movement.

12 stoplosses over 16 months at 20%, given your avg profit, heavily suggests lookahead or other problematic traits with your strategy.
